pennylane/qnode_new/jacobian.py

```diff
@@ -0,0 +1,528 @@
+
+            if not isinstance(op, CV):
+                return op.grad_method
+
+            # for CV ops it is more complicated
+            if op.grad_method == "A":
+                # op is Gaussian and has the heisenberg_* methods
+
+                obs_successors = self._op_descendants(op, 'E')
+                if not obs_successors:
+                    # op is not succeeded by any observables, thus analytic method is OK    FIXME actually we can ignore it....
+                    return 'A'
+
+                # check that all successor ops are also Gaussian
+                successor_ops = self._op_descendants(op, 'G')
+                if not all(x.supports_heisenberg for x in successor_ops):
+                    non_gaussian_ops = [x for x in successor_ops if not x.supports_heisenberg]
+                    # a non-Gaussian successor is OK if it isn't succeeded by any observables
+                    for x in non_gaussian_ops:
+                        if self._op_descendants(x, 'E'):
+                            return 'F'
+
+                # check successor EVs, if any order-2 observables are found return 'A2', else return 'A'
+                for observable in obs_successors:
+                    if observable.ev_order is None:
+                        # ev_order of None corresponds to a non-Gaussian observable
+                        return 'F'
+                    if observable.ev_order == 2:
+                        if observable.return_type is ObservableReturnTypes.Variance:
+                            # second order observables don't support
+                            # analytic diff of variances
+                            return 'F'
+                        op.grad_method = 'A2'  # bit of a hack
+                return 'A'
+            return op.grad_method
+
+        # operations that depend on this free parameter
+        ops = [d.op for d in self.variable_deps[idx]]
+        methods = list(map(best_for_op, ops))
+
+        if all(k in ('A', 'A2') for k in methods):
+            return 'A'
+
+        if None in methods:
+            return None
+
+        return 'F'
+
+    def jacobian(self, args, kwargs=None, *, wrt=None, method='B', options=None):
+        r"""Compute the Jacobian of the QNode.
+
+        Returns the Jacobian of the parametrized quantum circuit encapsulated in the QNode.
+        The Jacobian is returned as a two-dimensional array. The (possibly nested) input arguments
+        of the QNode are :func:`flattened <_flatten>` so the QNode can be interpreted as a simple
+        :math:`\mathbb{R}^m \to \mathbb{R}^n` function.
+
+        The Jacobian can be computed using several methods:
+
+        * Finite differences (``'F'``). The first order method evaluates the circuit at
+          :math:`n+1` points of the parameter space, the second order method at :math:`2n` points,
+          where ``n = len(wrt)``.
+
+        * Analytic method (``'A'``). Works for all one-parameter gates where the generator
```

Comment:
I think we should change `Analytic method` to `Parameter-shift method` (with corresponding change of the strings). The reason for this is because there may be other methods to compute analytic gradients (e.g., https://arxiv.org/abs/1905.13311), so we should keep flexible. We can update the docstring to make clear that parameter shift is analytic